File: rider_app/rolandrush_rider_flutter/lib/core/format.dart
import 'package:intl/intl.dart';

final _nairaFormat = NumberFormat.currency(locale: 'en_NG', symbol: '₦', decimalDigits: 0);

/// Ports utils/format.ts's naira().
String naira(num amount) => _nairaFormat.format(amount);

/// Ports utils/format.ts's nairaCompact().
String nairaCompact(num amount) {
  if (amount >= 1000) {
    final k = amount / 1000;
    final kStr = k % 1 == 0 ? k.toStringAsFixed(0) : k.toStringAsFixed(1);
    return '₦${kStr}k';
  }
  return '₦${amount.toStringAsFixed(0)}';
}

File: rider_app_ui_src/src/utils/format.ts
export function naira(amount: number): string {
  return `₦${amount.toLocaleString('en-NG')}`;
}

export function nairaCompact(amount: number): string {
  if (amount >= 1000) {
    const k = amount / 1000;
    return `₦${k % 1 === 0 ? k : k.toFixed(1)}k`;
  }
  return `₦${amount}`;
}